### PR TITLE
FIX: Selected ControlScheme lost upon entering PlayMode (ISXB-770)

### PR DESCRIPTION
**Pull request by timkeo**
### Description

Address an issue when entering PlayMode with the ActionsEditor window is open _and_ a ControlScheme is selected, i.e. something other than "All Control Schemes" is selected. The ControlScheme cannot be properly edited either during PlayMode or after returning to EditMode (due to various problems).

This occurs because the `InputControlScheme` object must be recreated after the domain reload.

### Changes made

Added logic to the overloaded `InputActionsEditorState` ctor to retrieve the selected ControlScheme's SerializedProperty (from the passed in SerializedObject) and initialize a new InputControlScheme() object.

The logic to detect and restore ActionsEditor state after a DR already existed, and simply needed to add the ControlScheme case.

### Notes

Also includes some simple formatting changes to `InputActionsEditorState`. Specifically, removed `this` from the field assignments: it's unnecessary (using m_ prefix for fields) and applied inconsistently.

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

File: Packages/com.unity.inputsystem/CHANGELOG.md
Comment (ekcoh) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: Nitpick but think it should be "Play Mode" since that is how its referred to in docs?

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorState.cs
Comment (ekcoh) [inline Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorState.cs]: Minor but would recommend keeping consistent if/else style here for readability and avoiding future mistakes so suggest keeping else clause also inside brackets and with line break similar to the if above.

**Review by ekcoh — APPROVED** [review]
Nice and clean, LGTM. Have to minor comments that I would like to be addressed before the merge but approving now since only minor/nitpicks.

**Review by Pauliusd01 — APPROVED** [review]
LGTM, tried editing schemes and their names during and after entering Play Mode, all of the changes are saved correctly with no errors.

**Review by ritamerkl — APPROVED** [review]

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorState.cs
Comment (timkeo) [reply Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorState.cs]: Oh yeah, that's one of my own personal styling quirks.
I tend to prefer having single-line else blocks on the same line as the else, but I'll be sure to use the correct package styling.

File: Packages/com.unity.inputsystem/CHANGELOG.md
Comment (timkeo) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: Fixed with amended commit (since needed to update branch anyway).

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorState.cs
Comment (timkeo) [reply Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorState.cs]: Fixed with amended commit.

**Comment by timkeo** [comment]
CI appears to be broken for test win/mac standalone on trunk builders with this error:
`Error building Player: Shader error in 'ProBuilder6/Standard Vertex Color': 'GLOBAL_CBUFFER_START': Too few arguments to a macro call. at /Users/bokken/build/output/Unity-Technologies/InputSystem/.Editor/Unity.app/Contents/CGIncludes/UnityShaderVariables.cginc(172)`
These failures also occur on develop branch, and so appears something broke on trunk.

NOTE: CI was green for this PR yesterday before updating from develop.

**Pull request merged**
